packages/detector.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Nathan Zender
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package packages
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+var execCommand = exec.Command
+
+func DetectPackageManager(verbose ...bool) (string, error) {
+	var os string
+	beChatty := len(verbose) > 0 && verbose[0] == true
+
+	installed := determineIfPackageManagerInstalled("yum", beChatty)
+	if installed {
+		//Having this be OS is a little weird. It probably should have been just package manager based flag.
```

Comment:
> //Having this be OS is a little weird. It probably should have been just package manager based flag.

Good point. Perhaps when this PR is updated for APK, this could be improved?